### DISCUSSION TITLE
PyMC NYC 2024 Sprint

### DISCUSSION THREAD

**Discussion by fonnesbeck** [pymc-devs/pymc3#7546] [General]
Welcome to the PyMC code sprint at PyData NYC 2024! We will use this discussion thread to highlight potential sprint projects and share information before, during and after the sprint.

## Candidate issues

Below are a list of beginner-friendly [issues](https://github.com/pymc-devs/pymc/issues) that you may want to consider. However, feel free to tackle any open issue if you think you are able to handle it. 

If you do choose an issue from this list, please add a checkmark so that others do not duplicate your effort!

### PyMC
- [ ] [Make multiprocessing import local to support pyodide #7519
](https://github.com/pymc-devs/pymc/issues/7519)
- [ ] [Deprecate SumTo1 Transform #7009](https://github.com/pymc-devs/pymc/issues/7009)
- [ ] [Implement logcdf for Truncated and Censored distributions #6686](https://github.com/pymc-devs/pymc/issues/6686)
- [ ] [DOC: progressbar_theme is undocumented #7468](https://github.com/pymc-devs/pymc/issues/7468)
- [ ] [Fix domain specification docstrings of ordered distributions #7486](https://github.com/pymc-devs/pymc/issues/7486)
- [ ] [Default target_accept not mentioned anywhere in the docs #7455](https://github.com/pymc-devs/pymc/issues/7455)
- [ ] [Add more information to ZeroSumNormal docs #7425](https://github.com/pymc-devs/pymc/issues/7425)
- [ ] [ENH: Disable all progress bars #7151](https://github.com/pymc-devs/pymc/issues/7151)

### PyTensor
- [ ] [Don't create useless FunctionGraph in linker tests #1022](https://github.com/pymc-devs/pytensor/issues/1022)
- [ ] [Extend inverse @ b as solve rewrite to batched dots #945](https://github.com/pymc-devs/pytensor/issues/945)

### PyMC Examples
- [ ] [Broken link in examples/generalized_linear_models/multilevel_modeling.ipynb #712](https://github.com/pymc-devs/pymc-examples/issues/712)
- [ ] [Bug in "Variational Inference: Bayesian Neural Networks" notebook #654](https://github.com/pymc-devs/pymc-examples/issues/654)
- [ ] [Broken link in MLDA intro page #620](https://github.com/pymc-devs/pymc-examples/issues/620)

---

We encourage everyone to use this discussion liberally to ask and answer questions related to the sprint!

**Comment by reshamas** [comment]
Here are some additional resources on contributing to PyMC:

a) Contributing to PyMC:
https://pymc-data-umbrella.xyz/en/latest/contributing/index.html

b) Video resources on contributing to PyMC:
https://www.dataumbrella.org/open-source/contributing-to-pymc

c) YouTube playlist of PyMC videos:
https://www.youtube.com/playlist?list=PLBKcU7Ik-ir99uTvN0315hIVLuyj4Q1Gt

**Comment by williambdean** [comment]
Other  "Beginner Friendly" issues to explore: 

- PyMC:  https://github.com/pymc-devs/pymc/labels/beginner%20friendly
- PyTensor: https://github.com/pymc-devs/pytensor/issues?q=is%3Aopen+is%3Aissue+label%3A%22beginner+friendly%22
- PyMC-Examples: https://github.com/pymc-devs/pymc-examples/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22

**Comment by wiwa** [comment]
Going to attempt https://github.com/pymc-devs/pytensor/issues/945

**Reply by bdice** [reply]
We're going to collaborate on this! 👍

**Comment by adithyalaks** [comment]
Attempting https://github.com/pymc-devs/pymc/issues/7519 !

**Reply by adithyalaks** [reply]
Unfortunately, I picked an issue that's probably unsolvable given that `multiprocessing` is in widespread use across PyMC samplers.

**Comment by takechanman1228** [comment]
FYI, I’m currently working on this issue and PR, thanks to @wd60622 .

https://github.com/pymc-labs/pymc-marketing/issues/1138
https://github.com/pymc-labs/pymc-marketing/pull/1172

**Comment by takechanman1228** [comment]
I've also tackled this issue: https://github.com/pymc-labs/pymc-marketing/issues/1137 and submitted a PR :  https://github.com/pymc-labs/pymc-marketing/pull/1173.
Thank you, @wd60622 

It was a great Sprint, thank you, everyone!
